### ISSUE TITLE
GT_Block_Ores_Abstract.onNeighborChange() event causes Stack Overflow

### ISSUE THREAD

**Issue by Ubus99** [GTNewHorizons/GT-New-Horizons-Modpack#15083]
### Your GTNH Discord Username

Ubus99

### Your Pack Version

2.4.0

### Your Server

private

### Java Version

Java 17

### Type of Server

Vanilla Forge

### Your Actions

Nothing specific, just walking around. Other players do not experience this error at the same time.
This happens to me from time to time, but in this specific session it happens every few minutes. Trying to restart and see if it stays that way

### Crash Report

https://mclo.gs/Kozz44K

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the crash still exists will prompt us to investigate and reopen it once we confirm your report.
- [x] I can reproduce this crash consistently by following the exact steps I described above.
- [x] I have asked other people and they confirm they also crash by following the exact steps I described above.

**Comment by Ubus99** [comment]
Not sure whether this is on the side of logistics pipes or GT, but it seems to be related to GT ores

**Comment by Ubus99** [comment]
Turns out this is not related to logistics pipes, it just happened that this event was triggered after logistics pipes a few times.
Still not able to find an existing issue, since this looks like something that should have been caught earlier. or there is a specific cause that i cant pin down yet

**Comment by bombcar** [comment]
https://discord.com/channels/181078474394566657/234936569360809996/1185608850963976242 - you mention it only happens from a particular chunk into another chunk - can you get us exact coordinates? Server side log at the same time it happens may also be useful and your full fml-latest.log (client side) when it occurs

**Comment by Ubus99** [comment]
Server log corresponding to the initial crash report, running from previous day until restart, time is two hours ahead of client.
[2023-12-16-1.log](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/13698216/2023-12-16-1.log)

Full client log:
[2023-12-16-1.log](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/13698295/2023-12-16-1.log)

Coordinates according to client crash-report footer, Z-direction seems to be important.
I don't see any report on the server, seems to be fully client-side.

We did some troubleshooting and I got another player to replicate the bug.
It did however disappear permanently after mining the black granite meteors located in the chunk that was being loaded by moving in the Z-Direction.
This points to a problem with gt ore rendering under very specific circumstances, since this only happens when approaching from that specific chunk, I was able to load the chunks containing black granite from any other direction as described in the discord conversation.

Not sure if this bug can be found with so little information, but it should at least point other people with similar issues in the right direction.
